src/main/scala/is/hail/io/reference/FASTAReader.scala

```diff
@@ -0,0 +1,185 @@
+package is.hail.io.reference
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+import java.util
+import java.util.Map.Entry
+
+import htsjdk.samtools.reference.{FastaSequenceIndex, FastaSequenceIndexCreator, ReferenceSequenceFile, ReferenceSequenceFileFactory}
+import is.hail.HailContext
+import is.hail.utils._
+import is.hail.variant.{RGBase, ReferenceGenome, Locus}
+import org.apache.hadoop
+
+import scala.language.postfixOps
+import scala.sys.process._
+
+class SerializableReferenceSequenceFile(val hConf: SerializableHadoopConfiguration, val fastaFile: String, val indexFile: String) extends Serializable {
+  @transient var value_ : ReferenceSequenceFile = _
+
+  def value: ReferenceSequenceFile = {
+    if (value_ == null) {
+      val localFastaFile = FASTAReader.getLocalFastaFileName(hConf, fastaFile, indexFile)
+      value_ = ReferenceSequenceFileFactory.getReferenceSequenceFile(new java.io.File(localFastaFile))
+    }
+    value_
+  }
+}
+
+object FASTAReader {
+  @volatile var localFastaFiles = Map.empty[String, String]
+
+  def getLocalIndexFileName(fastaFile: String): String =
+    ReferenceSequenceFileFactory.getFastaIndexFileName(new java.io.File(fastaFile).toPath).toString
+
+  def getLocalFastaFileName(hConf: SerializableHadoopConfiguration, fastaFile: String, indexFile: String): String = {
+    synchronized {
+      if (!localFastaFiles.contains(fastaFile)) {
+        val localFile = FASTAReader.setup(hConf, fastaFile, indexFile)
+        localFastaFiles += fastaFile -> localFile
+      }
+      localFastaFiles(fastaFile)
+    }
+  }
+
+  def setup(hConf: SerializableHadoopConfiguration, fastaFile: String, indexFile: String): String = {
```

Comment:
I think this method properly takes a `hadoop.conf.Configuration`. It doesn't need to serialize it, and taking a `SerializableHadoopConfiguration` adds the `.value` noise to the body of the method.

Reply:
It needs to be serializable because the FASTAReader class needs to be serializable and it needs the hadoop configuration to create the SerializableReferenceSequenceFile.

Reply:
Only the field needs to be serializable, I'm suggesting that we change this method signature and then change [this line](https://github.com/hail-is/hail/pull/3095/files/6bbff4c8379c26573a30f348c58970c26c423953#diff-7bb83d5b1df11e0472681a654860a1f2R37) to be:
```scala
val localFile = FASTAReader.setup(hConf.value, fastaFile, indexFile)
```
In particular, add `.value`.